src/Microsoft.AspNetCore.Authentication.Abstractions/AuthenticationProperties.cs

```diff
@@ -193,5 +188,55 @@ public string RedirectUri
                 }
             }
         }
+
+        /// <summary>
+        /// Gets or sets the error that happened during the authentication process.
+        /// </summary>
+        public string Error
+        {
+            get
+            {
+                return Items.TryGetValue(ErrorKey, out string value) ? value : null;
+            }
+            set
+            {
+                if (value != null)
```

Comment:
add a shared Get/SetHelper(key, value) to consolidate all of these?

Reply:
Maybe its worth adding a handful of tests for AuthenticationProperties since it looks like we don't have any today as well to verify the consolidated behavior

Reply:
Who will be setting these values?

Reply:
Me in the linked PR.

Reply:
Its effectively public API, we set them in auth, but all of the core IAuthenticationHandler apis also accept arbitrary AuthenticationProperties, its basically a settings bag that is passed along with authentication requests/results

Reply:
Actually this brings up a good point, this error logic is only needed for HandleRequestResult/RemoteHandlers to flow error info back right?  These properties should probably go into HandleRequestResult instead of AuthenticationProperties, since it looks like its only the RemoteHandler that would ever use this, and you can just store the error/description from HandleRemoteAuthenticateAsync here https://github.com/aspnet/Security/blob/9f7f8a857b258593bddf9067d740bb56d02bb90a/src/Microsoft.AspNetCore.Authentication/RemoteAuthenticationHandler.cs#L56